docs/src/operate/c8y/supported-operations.md

```diff
@@ -241,9 +243,177 @@ Arguments will be parsed correctly as long as following features are not include
+
+```json
+{
+  "delivery": {
+    "log": [],
+    "time": "2024-07-22T10:26:31.457Z",
+    "status": "PENDING"
+  },
+  "agentId":"201802315",
+  "creationTime":"2024-09-02T19:56:01.313Z",
+  "deviceId":"201802315",
+  "id":"1800380",
+  "status":"PENDING",
+  "c8y_Command":{
+    "text":"echo helloworld"
+  },
+  "description":"Execute command",
+  "externalSource":{
+    "externalId":"test-device",
+    "type":"c8y_Serial"
+  }
+}
+```
+
+The first approach assumes that user will create the operation file with the name of the operation:
+
+```sh
+sudo -u tedge touch /etc/tedge/operations/c8y/c8y_Command
+```
+
+... with the definition containing `on_fragment` field:
+
+```toml title="file: /etc/tedge/operations/c8y/c8y_Command"
+[exec]
+topic = "c8y/devicecontrol/notifications"
+on_fragment = "c8y_Command"
+command = "/etc/tedge/operations/command ${.payload.c8y_Command.text}"
+skip_status_update = true
+```
+
+The mapper is configured to pick up the message received on `topic` provided in the operation file field (if field is skipped, the default topic `c8y/devicecontrol/notifications` will be used). Mapper checks if `on_fragment` matches the one provided in the received message. Then the operation plugin located at `/etc/tedge/operations/command` would be executed with the arguments provided in message payload (in this example `${.payload.c8y_Command.text}` will be replaced with `echo helloworld`).
+
+:::info
+Users can provide their own custom topic in the operation file:
+
+```toml title="file: /etc/tedge/operations/c8y/c8y_Command"
+[exec]
+topic = "${.bridge.topic_prefix}/custom/topic"
+on_fragment = "c8y_Command"
+command = "/etc/tedge/operations/command ${.payload.c8y_Command.text}"
+skip_status_update = true
+```
+
+Instead of providing `topic_prefix` manually, you can use `${.bridge.topic_prefix}` template to derive it from the bridge config of current cloud instance.
+
+The mapper will automatically subscribe to all custom topic provided in operation file under `topic` field when connecting to the cloud.  
+:::
+
+:::note
+When `on_fragment` is provided with `on_message`, the mapper will ignore the operation.
+:::
+
+For the second approach, the user is supposed to create a master custom operation definition file with `.template` extension, e.g.:
+
+```toml title="file: /etc/tedge/operations/c8y/c8y_Command.template"
+[exec]
+topic = "c8y/devicecontrol/notifications"
+on_fragment = "c8y_Command"
+
+[exec.workflow]
+operation = "command"
+input = "${.payload.c8y_Command.text}"
+```
+
+To indicate that devices are supporting the operation, the device must publish a command capability message where command name is derived from `exec.workflow.operation`:
+
+```sh
+tedge mqtt pub -r 'te/device/<name>///cmd/command' '{}'
+```
+
```

Comment:
Maybe it's better to add a note that says the capability message is sent by workflow automatically once the workflow file with the same command name is defined.

Reply:
Resolved by a267f04581ec93b1644543df4d45e998b590aaa8